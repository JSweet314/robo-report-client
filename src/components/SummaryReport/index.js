import React from 'react';
import PropTypes from 'prop-types';

const SummaryReport = ({ 
  questionBlocks, 
  values
}) => {
  const reportItems = questionBlocks.map(block => block.questions)
    .reduce((reportItems, questions) => [...reportItems, ...questions], []);

  const summary = reportItems.map(question => {
    const value = values[question.value] === '-' || !values[question.value] ?
      'N/A' : values[question.value];
    return (
      <p key={question.value}>
        <strong>{question.label}</strong>: {value}
      </p>
    );
  });

  return (
    <div>
      { summary }
    </div>
  );
};

SummaryReport.propTypes = {
  values: PropTypes.object.isRequired,
  questionBlocks: PropTypes.array.isRequired
};

export default SummaryReport;