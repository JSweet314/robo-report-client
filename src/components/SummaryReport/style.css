.summary__item {
  margin: 0.5rem 0;
}

.summary_question {
  display: block;
}