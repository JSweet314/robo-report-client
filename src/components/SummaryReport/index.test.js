import React from 'react';
import { shallow } from 'enzyme';
import SummaryReport from './index';
import questionBlocks 
  from '../../containers/NewComplaintContainer/complaintQuestions';

describe('SummaryReport', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <SummaryReport 
        values={{}}
        questionBlocks={questionBlocks}
      />
    );
  });

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});