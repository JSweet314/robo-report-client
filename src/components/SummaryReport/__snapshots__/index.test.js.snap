// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryReport should match a snapshot 1`] = `
<div>
  <p
    key="subject"
  >
    <strong>
      Subject
    </strong>
    : 
    N/A
  </p>
  <p
    key="description"
  >
    <strong>
      Description
    </strong>
    : 
    N/A
  </p>
  <p
    key="isSoliciting"
  >
    <strong>
      Did the call/message that you are reporting advertise any type of property, goods, or services?
    </strong>
    : 
    N/A
  </p>
  <p
    key="typeOfSolicit"
  >
    <strong>
      Type of Property, Goods, or Services
    </strong>
    : 
    N/A
  </p>
  <p
    key="doneBusinessWith"
  >
    <strong>
      Have you or anyone else in your household done any business with the caller / company within the past 18 months immediately before you received the call / message?
    </strong>
    : 
    N/A
  </p>
  <p
    key="inquiredWith"
  >
    <strong>
      Have you or anyone else in your household made any inquiry or application to the caller/company within the 3 months immediately before you received the call/message?
    </strong>
    : 
    N/A
  </p>
  <p
    key="householdRelation"
  >
    <strong>
      Do you or anyone in your household have a personal relationship with the individual that made the call?
    </strong>
    : 
    N/A
  </p>
  <p
    key="permissionToCall"
  >
    <strong>
      Have you or anyone else in your household or business given the caller/company permission to call?
    </strong>
    : 
    N/A
  </p>
  <p
    key="writtenPermission"
  >
    <strong>
      Did you give permission to the caller/business to call you in writing?
    </strong>
    : 
    N/A
  </p>
  <p
    key="dateOfPermission"
  >
    <strong>
      Date you provided permission to the caller/company to call
    </strong>
    : 
    N/A
  </p>
  <p
    key="date"
  >
    <strong>
      Date of Your Issue/Problem
    </strong>
    : 
    N/A
  </p>
  <p
    key="time"
  >
    <strong>
      Time of Your Issue/Problem
    </strong>
    : 
    N/A
  </p>
  <p
    key="typeOfCall"
  >
    <strong>
      Type of Call or Message
    </strong>
    : 
    N/A
  </p>
  <p
    key="receivedCallerId"
  >
    <strong>
      Did you receive Caller ID Information?
    </strong>
    : 
    N/A
  </p>
  <p
    key="callerIdNumber"
  >
    <strong>
      Caller ID Number
    </strong>
    : 
    N/A
  </p>
  <p
    key="callerIdName"
  >
    <strong>
      Caller ID Name
    </strong>
    : 
    N/A
  </p>
  <p
    key="receivedBusinessName"
  >
    <strong>
      Was the caller's business name provided during the call/message?
    </strong>
    : 
    N/A
  </p>
  <p
    key="nameAtBeginning"
  >
    <strong>
      Was the business name provided at the beginning of the call/message?
    </strong>
    : 
    N/A
  </p>
  <p
    key="providedAdvertiserName"
  >
    <strong>
      Please provide the name of the advertiser provided during the call
    </strong>
    : 
    N/A
  </p>
  <p
    key="providedAdvertiserNumber"
  >
    <strong>
      Provide the advertiser's phone number given during the call.
    </strong>
    : 
    N/A
  </p>
</div>
`;
